### PR TITLE
Feat: consolidate bbox transforms

### PR DESCRIPTION
**Pull request by liyier90**
Closes https://github.com/aimakerspace/PeekingDuck-Private/issues/28
Closes https://github.com/aimakerspace/PeekingDuck-Private/issues/64

- Move common bbox conversion functions to `peekingduck/pipeline/utils/bbox/transforms.py`
- Standardize the naming of bboxes in PeekingDuck to be:
  - tlwh (t, l, w, h): (t, l) is top left corner, width and height. Mostly used by tracking models
  - xyxy (x1, y1, x2, y2):  (x1, y1) is top left and (x2, y2) is bottom right
  - xyxyn (x1, y1, x2, y2): Normalized xyxy. Relative coordinates.
  - xywh (x, y, w, h): (x, y) is center coord, width and height
  - xyah (x, y, a, h): (x, y) is center coord, aspect ratio and height
- Refactor code to use these common bbox conversion functions where possible
- Removed unnecessary parameter conversion in MTCNN, assigned the converted parameters to member variable in the constructor instead.

Note: `xywh` is also called `cxcywh` [example](https://github.com/MarkAny-Vision-AI/CenterNeXt/blob/0ee660b9ab6d85d12b128654624779beca2b9c2d/utils/data/transforms.py#L7) but `xywh` gave more hits on GitHub suggesting it's more commonly used.

### CODE REVIEW

File: peekingduck/pipeline/utils/bbox/transforms.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    """Converts from [x1, y1, x2, y2] to normalized [x1, y1, x2, y2].
+
+    (x1, y1) is the top left corner and (x2, y2) is the bottom right corner.
+    Normalized coordinates are w.r.t. original image size.
+    """
+    outputs = np.empty_like(inputs)
+    outputs[:, [0, 2]] = inputs[:, [0, 2]] / width
+    outputs[:, [1, 3]] = inputs[:, [1, 3]] / height
+
+    return outputs
```
Comment (ongtw) [inline peekingduck/pipeline/utils/bbox/transforms.py:96]: For all the above functions:
1. Would be instructive to users if docstrings could include a numeric example showing the input and output.
2. For each x, y, w, h, t, l : if there are expectations for any to be in the range (0, 1) (relative vs abs coords), would be good to have assert statements to check for input range errors.
3. Some functions (e.g. `tlwh2xyah`) accept single dimension input, e.g. [0, 0, 1, 1], while others (e.g. `tlwh2xyxyn`) require multi-dimension input, e.g. [[0, 0, 1, 1]]. Would be good to document this somewhere in the docstrings.

**Review by ongtw — CHANGES_REQUESTED** [review]
Some suggestions for your consideration pls.

File: peekingduck/pipeline/utils/bbox/transforms.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def tlwh2xyxyn(inputs: np.ndarray, height: int, width: int) -> np.ndarray:
+    """Converts from [t, l, w, h] to [x1, y1, x2, y2] format.
+
+    (x1, y1) and (x2, y2) are coordinates of top left and bottom right
+    respectively. (t, l) is the coordinates of the top left corner, w is the
+    width, and h is the height.
+    """
```
[the remark this answers is not in the document]
Comment (liyier90) [reply peekingduck/pipeline/utils/bbox/transforms.py:83]: Added equation to show how the output is calculated and added example input and output of the function in the docstrings

File: peekingduck/pipeline/utils/bbox/transforms.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    Returns:
+        (np.ndarray): Bounding box with (x, y, a, h) format.
+    """
+    outputs = np.asarray(inputs).copy()
+    outputs[:2] += outputs[2:] / 2
+    outputs[2] /= outputs[3]
+    return outputs
+
+
+def tlwh2xyxyn(inputs: np.ndarray, height: int, width: int) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (liyier90) [reply peekingduck/pipeline/utils/bbox/transforms.py:55]: added explanation that the output is normalized coordinates

File: peekingduck/pipeline/utils/bbox/transforms.py (hunk 1 above)
Comment (liyier90) [reply peekingduck/pipeline/utils/bbox/transforms.py:186]: 1. Added example input/output for each function
2. For now, these functions have not expectation for input to be of a specific range. Will use assert statements if necessary in the future
3. Specified in the `Args` section of the docstring that `inputs` is either a 1-d array or 2-d array.

**Review by ongtw — APPROVED** [review]
The updated docstrings with equations and examples are very nice: they really help to explain each function.

**Pull request merged**